src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,143 @@
 import React from 'react';
+import classNames from 'classnames';
+import { getTodos } from '../../api';
+
 import './TodoList.scss';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+interface Props {
+  selectedUserId: number;
+  changeUserId: (userId: number) => void;
+}
+
+interface State {
+  todos: Todo[] | [];
+  value: string;
+  filterType: string;
+}
+
+export class TodoList extends React.Component<Props, State> {
+  state: State = {
+    todos: [],
+    value: '',
+    filterType: 'all',
+  };
+
+  componentDidMount() {
+    getTodos()
+      .then((items: Todo[]) => {
+        this.setState({
+          todos: items,
+        });
+      });
+  }
+
+  hundleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      value: event.target.value,
+    });
+  };
+
+  filterActive = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState({
+      filterType: event.target.value,
+    });
+  };
+
+  getFilteredTodos = () => {
+    const { todos, value, filterType } = this.state;
+
+    let filteredTodos = todos.filter((todo: Todo) => (
+      todo.title
+      && todo.title.toLowerCase()
+        .includes(value.toLowerCase())
+    ));
```

Comment:
```suggestion
    let filteredTodos = todos.filter((todo: Todo) => (
      todo.title
      && todo.title.toLowerCase().includes(value.toLowerCase())
    ));
```

for value.toLowerCase() better create a separate variable not to calculate it on every iteration